subsys/bluetooth/controller/hal/nrf5/radio/radio.c

```diff
@@ -5,7 +5,16 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 #include <soc.h>
+#if !defined(CONFIG_ARCH_POSIX)
 #include <arch/arm/cortex_m/cmsis.h>
+#endif
+
+#if defined(CONFIG_BOARD_NRF52_NWTSIM)
```

Comment:
General comment regarding the conditional-compilation for simulator: could this macro name be more generic, to be able to accommodate boards of more nRF SOCs in the future, for example, BOARD_NRF52840_NWTSIM or even BOARD_NRF51_NWTSIM?

E.g would it make sense to be named: CONFIG_BOARD_NRF5X_NWTSIM?

Of course, I suppose that these new boards may require different HW models, but this is something that can be solved internally in the HW model function (e.g. with #ifdef 52, #ifdef 52840, etc.)

Reply:
Certainly. We can just call it CONFIG_BOARD_NRFXX_NWTSIM.